crates/staking/src/lib.rs

```diff
@@ -25,6 +25,12 @@ pub(crate) type SignedFixedPoint<T> = <T as Config>::SignedFixedPoint;
 
 pub use pallet::*;
 
+pub type DefaultVaultId<T> = VaultId<<T as frame_system::Config>::AccountId, <T as Config>::CurrencyId>;
```

Comment:
This can definitely be generic on the `Config` trait in this pallet, then no changes are required in the tests. You could have to bind `Into<AccountId>` for it though which I think you mentioned there was a problem with?

Reply:
You mean, to make `VaultId an associated type in the config trait?

Reply:
Yes what you want is not possible without changing the type used as account id, which defeats the purpose. You indeed need something like `Into<AccountId>`, and you can't implement that on `u64` in mock.rs due to the orphan rule